obspy/taup/tau.py

```diff
@@ -83,6 +84,55 @@ def __init__(self, arrivals, model):
         self.model = model
         self.extend(arrivals)
 
+    def __add__(self, other):
+        if isinstance(other, Arrival):
+            other = Arrivals([other], model=self.model)
+        if not isinstance(other, Arrivals):
+            raise TypeError
+        return self.__class__(super(Arrivals, self).__add__(other),
+                              model=self.model)
+
+    def __iadd__(self, other):
+        if isinstance(other, Arrival):
+            other = Arrivals([other], model=self.model)
+        if not isinstance(other, Arrivals):
+            raise TypeError
+        self.extend(other)
+        return self
+
+    def __mul__(self, num):
+        if not isinstance(num, int):
+            raise TypeError("Integer expected")
+        arr = self.copy()
+        for _i in range(num-1):
+            arr += self.copy()
+        return arr
+
+    def __setitem__(self, index, arrival):
+        if isinstance(arrival, Arrival):
+            super(Arrivals, self).__setitem__(index, arrival)
+        else:
+            msg = 'Only Arrival objects can be assigned.'
+            raise TypeError(msg)
+
+    def __setslice__(self, i, j, seq):
+        if all(isinstance(x, Arrival) for x in seq):
+            super(Arrivals, self).__setslice__(i, j, seq)
+        else:
+            msg = 'Only Arrival objects can be assigned.'
+            raise TypeError(msg)
+
+    def __getitem__(self, index):
+        if isinstance(index, slice):
+            return self.__class__(super(Arrivals, self).__getitem__(index),
+                                  model=self.model)
+        else:
+            return super(Arrivals, self).__getitem__(index)
+
+    def __getslice__(self, i, j):
```

Comment:
The problem is that, in Python 2.7.12, if I remove the `__getslice__()` method, slicing returns a `list` object:
```python
In [3]: type(arrivals[1:])
Out[3]: list
```

(and I saw that we [do have](https://github.com/obspy/obspy/blob/master/obspy/core/stream.py#L665) a `__getslice__()` method for the `Stream` class.